kbase-extension/static/kbase/js/widgets/function_output/kbaseSampleSet.js

```diff
@@ -0,0 +1,228 @@
+/*
+Sebastian Le Bras - April 2020
+*/
+define ([
+    'kbwidget',
+    'bootstrap',
+    'jquery',
+    'kbase-client-api',
+    'widgets/dynamicTable',
+    'kbaseAuthenticatedWidget',
+    'kbaseTabs',
+    'kbase-generic-client-api',
+    'narrativeConfig',
+    'bluebird'
+], function(
+	KBWidget,
+	bootstrap,
+	$,
+    kbase_client_api,
+    DynamicTable,
+	kbaseAuthenticatedWidget,
+    kbaseTabs,
+    GenericClient,
+    Config,
+    Promise
+) {
+    'use strict';
+
+    return KBWidget({
+        name: 'kbaseSampleSetView',
+        parent: kbaseAuthenticatedWidget,
+        version: '1.0.0',
+        options: {
+            pageLimit: 10,
+            default_blank_value: ""
+        },
+
+        init: function init(options) {
+            this._super(options);
+
+            var self = this;
```

Comment:
Using `self` vs `this` mostly isn't necessary here. That's mainly to handle scope, but like 90% of this function has the main widget (`this`) in the scope you expect.

The only place that it's different is in the response to `get_objects2`, but if you pass in an arrow function, that should preserve scope.